src/vs/vscode.proposed.d.ts

```diff
@@ -33,7 +33,37 @@ declare module 'vscode' {
 		extensionHostEnv?: { [key: string]: string | null };
 	}
 
-	export type ResolverResult = ResolvedAuthority & ResolvedOptions;
+	export interface Port extends Disposable {
+		remotePort: number;
+		localPort?: number;
+		description?: string;
+	}
+
+	/**
+	 * Used as part of the ResolverResult if the extension has any candidate, published, or forwarded ports.
+	 */
+	export interface PortInformation {
+		/**
+		 * Ports the extension knows are interesting and might be nice to forward.
+		 */
+		candidates?: Port[];
+		/**
+		 * Ports that are already immutably forwarded.
+		 */
+		published?: Port[];
+		/**
+		 * An array of ports to be forwarded once connected. For example,
+		 * if the extension wants to have a configuration file with some default ports, they could be passed through for forwarding here.
+		 */
+		toForward?: { remote: number, local: number, name: string }[];
```

Comment:
The properties are slightly different. I'm not sure that `Port` needs a name, because that's something only the user can provide (either through new UI or through a config file). I also added a `closeable` property here, which I don't see a need for on `Port` right now.